libgpos/src/task/CAutoTaskProxy.cpp

```diff
@@ -506,18 +506,14 @@ CAutoTaskProxy::Execute
 	{
 		// mark task as erroneous
 		ptsk->SetStatus(CTask::EtsError);
-
-		if (m_fPropagateError)
-		{
-			GPOS_RETHROW(ex);
-		}
+		GPOS_RETHROW(ex);
```

Comment:
In this case, you ignore the flag `m_fPropagateError` and always raise the exception regardless.

If you change the `_api.cpp` from `false` to `true`, is that good enough?

I see we lost a code path where we DO want to eat the exceptions.